ring/replication_set_test.go

```diff
@@ -564,6 +564,78 @@ func TestDoUntilQuorumWithoutSuccessfulContextCancellation_PartialZoneFailure(t
 	}
 }
 
+func TestDoUntilQuorumWithoutSuccessfulContextCancellation_CancelsEntireZoneImmediatelyOnSingleFailure(t *testing.T) {
```

Comment:
that name tho :-o

Reply:
`DoUntilQuorumWithoutSuccessfulContextCancellation` is deliberately named to be awkward 🙂 Most consumers should use the far more friendly-sounding `DoUntilQuorum` unless they need the different behaviour the unwieldy-named method provides.